src/contracts/atlas/Factory.sol

```diff
@@ -30,10 +30,17 @@ abstract contract Factory {
     /// @notice Creates a new Execution Environment for the caller, given a DAppControl contract address.
     /// @param control The address of the DAppControl contract for which the execution environment is being created.
     /// @return executionEnvironment The address of the newly created Execution Environment instance.
-    function createExecutionEnvironment(address control) external returns (address executionEnvironment) {
+    function createExecutionEnvironment(
+        address control,
+        address user
+    )
+        external
+        returns (address executionEnvironment)
+    {
+        if (msg.sender != control && msg.sender != user) revert AtlasErrors.Unauthorized();
```

Comment:
Should that condition be an or otherwise we wouldn't need 2 checks.
msg.sender is either control or user

Reply:
If it's an `OR` and `msg.sender == user`, wouldn't the lefthand of the `OR` fail and immediately execute the revert?

Reply:
Actually yes we want the msg.sender to be neither `user` nor `control`.
I initially thought it was the opposite

Reply:
Think the logic is fine. But it might be more gas efficient to put the `msg.sender != user` check first, as that is the more likely scenario, so we can evaluate to `false` early and skip the 2nd check.
```suggestion
        if (msg.sender != user && msg.sender != control) revert AtlasErrors.Unauthorized();
```